WalletWasabi.Fluent/ViewModels/Wallets/CoinJoinStateViewModel.cs

```diff
@@ -89,7 +92,19 @@ public CoinJoinStateViewModel(WalletViewModel walletVm, IObservable<Unit> balanc
 
 		balanceChanged.Subscribe(_ => _stateMachine.Fire(Trigger.BalanceChanged));
 
-		PlayCommand = ReactiveCommand.Create(() => _stateMachine.Fire(Trigger.Play));
+		PlayCommand = ReactiveCommand.Create(async () =>
+		{
+			bool changed = false;
+			if (!_wallet.KeyManager.IsCoinjoinProfileSelected)
+			{
+				DialogResult<bool> isProfileSelected = await NavigateDialogAsync(new CoinJoinProfilesViewModel(_wallet.KeyManager, false), NavigationTarget.DialogScreen);
+				changed = isProfileSelected.Result;
+			}
+			if (changed || _wallet.KeyManager.IsCoinjoinProfileSelected)
```

Comment:
Is `changed` required here? What does `changed` do?

Reply:
It can be removed and be
```
			if (!_wallet.KeyManager.IsCoinjoinProfileSelected)
			{
				DialogResult<bool> isProfileSelected = await NavigateDialogAsync(new CoinJoinProfilesViewModel(_wallet.KeyManager, false), NavigationTarget.DialogScreen);
			}
			if (_wallet.KeyManager.IsCoinjoinProfileSelected)
			{
				_stateMachine.Fire(Trigger.Play);
			}
```
If we change the second if to an else, the first time we click on it to select a profile, and we do select, we need to click on Play again for it to start.